src/providers/index.tsx

```diff
@@ -1,7 +1,23 @@
-/*
- *
- * @provider AuthProvider
- *
- */
+export { ApplicationsProvider, useApplications } from "./ApplicationsProvider";
 
-export { AuthProvider } from "./auth.provider";
+export { AuthProvider, useUser, useAuth } from "./AuthProvider";
+export type {
+    UserWithClaims,
+    UserType,
+    ProviderName,
+    AuthMethod,
+    AuthContextValue,
+} from "./AuthProvider";
+
+export {
+    RoutesProvider,
+    useRouter,
+    useHeaderInfo,
+    useRouteDefinitions,
+} from "./RoutesProvider";
+export type {
+    HeaderInfo,
+    RoutesContextValue,
+    PathObject,
+    RouteConfig,
+} from "./RoutesProvider";
```

Comment:
This is a bit subjective, but I don't find index files particularly helpful in the context of an app. You can still circumvent the index file and import directly from the respective files, now there's two ways of importing things. You also have to be particularly careful to avoid import cycles. Plus, in large projects, barrel files start to have a significant impact on performance of tools like ESLint or Vitest due to having to load every file in the index file instead of just the one you are interested in.

Here's an article written by TkDodo, author of tanstack query, on why index/barrel files are bad.
https://tkdodo.eu/blog/please-stop-using-barrel-files

In a small project, it doesn't matter as much, but it's not a good practice all the same.

Reply:
Thanks for bringing this up. We need a massive review on the current code organization since it was rushed for last year and many decisions were badly made. I am very indifferent when it comes to using index files or not. I am down to do the switch because populating the index file is tiresome ngl.